### PR TITLE
Fixes #29792  - Add current taxonomies and user in react context

### PR DESCRIPTION
**Pull request by amirfefer**
current loc / org and user-id should be a single source of truth across core and plugins and react context may be the place. 
The need for this PR has been raised here:
https://github.com/theforeman/foreman_plugin_template/pull/32/files#diff-2ee0f3275ce16ab586fcec80d0b2124cR11

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -517,7 +517,12 @@ def current_url_params(permitted: [])
   end
 
   def app_metadata
-    { UISettings: ui_settings, version: SETTINGS[:version].short, docUrl: documentation_url }
+    {
+      UISettings: ui_settings, version: SETTINGS[:version].short, docUrl: documentation_url,
+      location: Location.current&.id,
+      organization: Organization.current&.id,
+      user: User.current&.id
```
Comment (xprazak2) [inline app/helpers/application_helper.rb]: nit: How about `locationId`, `organizationId`, `userId` for the keys?

**Comment by xprazak2** [comment]
Looks good, I just added one comment.

**Review by tbrisker — COMMENTED** [review]
How will we handle changing the current taxonomy in the UI (right now it won't be a problem due to triggering full page reload, but as pages change to react it might), as well as impersonation?

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (tbrisker) [reply app/helpers/application_helper.rb]: Should we make these into objects that also contain the taxonomy label or the user full name so they can be used for display purposes?

**Comment by tbrisker** [comment]
what's the status here @amirfefer ?

**Comment by ofedoren** [comment]
@amirfefer needs rebase :(

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (amirfefer) [reply app/helpers/application_helper.rb]: makes sense

**Comment by amirfefer** [comment]
@tbrisker, @xprazak2, rebased, plus I  added a few changes.
> How will we handle changing the current taxonomy in the UI (right now it won't be a problem due to triggering full page reload, but as pages change to react it might), as well as impersonation?


Changing a taxonomy triggers a full-page reload, including impersonation, so the context is updated.
React context is mutable via the client-side, so in the future, we can still use this pattern.

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -500,7 +500,12 @@ def current_url_params(permitted: [])
   end
 
   def app_metadata
-    { UISettings: ui_settings, version: SETTINGS[:version].short, docUrl: documentation_url }
+    {
+      UISettings: ui_settings, version: SETTINGS[:version].short, docUrl: documentation_url,
+      location: Location.current.attributes.slice('id', 'title'),
+      organization: Organization.current.attributes.slice('id', 'title'),
```
Comment (xprazak2) [inline app/helpers/application_helper.rb]: Needs a check if current loc/org is set, otherwise fails on undefined method when in 'Any Context'.

Comment (amirfefer) [reply app/helpers/application_helper.rb]: Thanks @xprazak2, should work now

**Review by xprazak2 — APPROVED** [review]

**Comment by xprazak2** [comment]
Thanks!

**Pull request merged**